Add PHP Metrics docs (#8972)
getsentry/sentry-docs: src/platforms/php/common/metrics/index.mdx

---

> title: Metrics

Reviewer: ```suggestion
title: Set Up Metrics
```

description: "Learn how to configure Metrics in your app."
---

> <Note>
> 
> Metrics for PHP are supported with Sentry PHP SDK version `4.5.0` and above.
> 
> </Note>
> 

Reviewer 2: ```suggestion
<Note>

Metrics for PHP are supported with Sentry PHP SDK version `4.5.0` and above.

</Note>

Sentry metrics help you pinpoint and solve issues that impact user experience and app performance by measuring the data points that are important to you. You can track things like processing time, event size, user signups, and conversion rates, then correlate them back to tracing data in order to get deeper insights and solve issues faster.
```

It's good to give users a bit of an intro/user benefit :)

> ## Setup

Reviewer: ```suggestion
## Set Up Code Location
```

> 
> ```php
> \Sentry\init([
>    'dsn' => '___PUBLIC_DSN___',
>    'attach_metric_code_locations' => true, 
> ]);
> ```

Reviewer 2: ```suggestion

```

> To attach the code location from where your metrics were emitted from, set `attach_metric_code_locations` to `true` (default is `false`).

Reviewer 2: ```suggestion
To attach the code location of your metrics, set `attach_metric_code_locations` to `true` (the default is `false`).

```php
\Sentry\init([
   'dsn' => '___PUBLIC_DSN___',
   'attach_metric_code_locations' => true, 
]);
```
```

> ## Emit a counter

Reviewer 2: ```suggestion
## Emit a Counter
```

> Counters are the simplest type of metrics. With counters, you can count occurrences of certain events.

Reviewer 2: ```suggestion
Counters are one of the more basic types of metrics and can be used to count certain event occurrences.

To emit a counter, do the following:
```

```php
// Increment a counter by one for each button click.
\Sentry\metrics()->increment(
    key: 'button_click',
    value: 1,
    tags: [
        'browser' => 'Firefox',
        'app_version' => '1.0.0',
    ],
)
```

> ## Emit a distribution

Reviewer 2: ```suggestion
## Emit a Distribution
```

> 
> Distributions are useful to get the most insights from your data. With distributions, you can obtain aggregations such as `p90`, `min`, `max`, and `avg`, making them very powerful.
> 

Reviewer 2: ```suggestion

Distributions help you get the most insights from your data by allowing you to obtain aggregations such as `p90`, `min`, `max`, and `avg`.

To emit a distribution, do the following:

```

```php
// Add '15.0' to a distribution used for tracking the loading times per page.
\Sentry\metrics()->distribution(
    key: 'page_load',
    value: 15.0,
    unit: \Sentry\Metrics\MetricsUnit::millisecond(),
    tags: [
        'page' => '/home',
    ],
)
```

> ## Emit a set
> 
> Sets are useful for counting unique occurrences. With sets, you can obtain the count of unique elements you added, making them very useful for certain cases.
> 

Reviewer 2: ```suggestion
## Emit a Set

Sets are useful for looking at unique occurrences and counting the unique elements you added.

To emit a set, do the following:

```

```php
// Add 'jane' to a set used for tracking the number of users that viewed a page.
\Sentry\metrics()->set(
    key: 'user_view',
    value: 'jane',
    unit: \Sentry\Metrics\MetricsUnit::custom('username'),
    tags: [
        'page' => '/home',
    ],
)
```

> ## Emit a gauge
> 
> Gauges are a special kind of metric. You can obtain aggregates like `min`, `max`, `avg`, `sum`, and `count` with gauges but no percentiles. You might wonder why I need gauges if I have distributions that can do the same and more. The answer has to do with space efficiency since gauges can be represented in a much more space-efficient way, so if you don't need percentiles, gauges are the way to go.
> 

Reviewer 2: ```suggestion
## Emit a Gauge

Gauges let you obtain aggregates like `min`, `max`, `avg`, `sum`, and `count`. They can be represented in a more space-efficient way than distributions, but they can't be used to get percentiles. If percentiles aren't important to you, we recommend using gauges.

To emit a gauge, do the following:

```

```php
// Add '15.0' to a gauge used for tracking the loading times for a page.
\Sentry\metrics()->set(
    key: 'page_load',
    value: 15.0,
    unit: \Sentry\Metrics\MetricsUnit::millisecond(),
    tags: [
        'page' => '/home',
    ],
)
```

> ## Emit a timer
> 
> Timers, which are implemented as a distribution with unit second, can be used to measure the execution time of a certain block of code.
> 

Reviewer 2: ```suggestion
## Emit a Timer

Timers can be used to measure the execution time of a specific block of code. They're implemented like distributions, but measured in seconds.

To emit a timer, do the following:

```

```php
\Sentry\metrics()->timing(
    key: 'event_processing_time',
    callback: static fn() => process(),
);
```

> ## Flush your metrics
> 
> All metrics emitted need to be manually flushed to Sentry.

Reviewer 2: ```suggestion
## Flush Your Metrics

All metrics emitted have to be manually flushed to Sentry.

To flush your metrics, do the following:
```
